### PR TITLE
fix shell check error for case-lib/*

### PR DESCRIPTION
**Pull request by Bin-QA**
Update for fix shell check error for case-lib/*
expect case-lib/config.sh file

### CODE REVIEW

**Comment by Bin-QA** [comment]
Wait for CI system detect error

File: case-lib/lib.sh [hunk 1]
```diff
@@ -1,43 +1,70 @@
...
+export DMESG_LOG_START_LINE
 
-declare -g SOF_LOG_COLLECT=0
+export SOF_LOG_COLLECT=0
 
 func_lib_setup_kernel_last_line()
 {
-    declare -g KERNEL_LAST_LINE=$(wc -l /var/log/kern.log|awk '{print $1;}')
+    KERNEL_LAST_LINE=$(wc -l /var/log/kern.log|awk '{print $1;}')
+    export KERNEL_LAST_LINE
```
Comment (marc-hb) [inline case-lib/lib.sh:56]: See above. `declare -g` is a no-op, drop it (`export` is `declare -x`, not `declare -g`)

File: case-lib/lib.sh [hunk 2]
```diff
@@ -1,43 +1,70 @@
...
 
 func_lib_start_log_collect()
 {
-    local is_etrace=${1:-0}
-    local ldcFile=/etc/sof/sof-$(sof-dump-status.py -p).ldc
+    local is_etrace=${1:-0} ldcFile
+    ldcFile=/etc/sof/sof-$(sof-dump-status.py -p).ldc || {
+        dlogw "sof-dump-status.py -p to query platformat failed"
+        return
+    }
```
Comment (marc-hb) [inline case-lib/lib.sh:65]: This is useful and generally not visible externally but it's unfortunate to hide this in a commit with a placeholder commit message.

File: case-lib/lib.sh [hunk 3]
```diff
@@ -46,39 +73,38 @@ func_lib_start_log_collect()
...
 
 func_lib_check_sudo()
 {
-    func_hijack_setup_sudo_level
-    [[ $? -ne 0 ]] && \
-        dlogw "Command needs root privilege to run, please configure SUDO_PASSWD in case-lib/config.sh" && \
+    func_hijack_setup_sudo_level || {
+        dlogw "Command needs root privilege to run, please configure SUDO_PASSWD in case-lib/config.sh"
         exit 2
+    }
```
Comment (marc-hb) [inline case-lib/lib.sh:88]: Use a `die()` function

File: case-lib/logging_ctl.sh [hunk 4]
```diff
@@ -27,52 +26,41 @@ _func_log_cmd()
...
+    ppcmd=$(ps -p $PPID -o cmd --noheader|awk '{print $2;}')
+    local ext_message="" cmd
+    # confirm this script is loaded by other script, and add the flag for it
+    file "$ppcmd" 2>/dev/null |grep -q 'shell script' && ext_message=" Sub-Test:"
+    for key in "${!LOG_LIST[@]}";
+    do
+        # dymaic alias the command with different content
+        cmd="alias $key='echo \$(date -u \"+%F %T %Z\")$ext_message ${LOG_LIST[$key]}'"
+        eval "$cmd"
+    done
```
Comment (marc-hb) [inline case-lib/logging_ctl.sh:41]: Please explain this change in a different commit.

File: case-lib/logging_ctl.sh [hunk 5]
```diff
@@ -27,52 +26,41 @@ _func_log_cmd()
...
-    local log_dir=$(dirname ${BASH_SOURCE[0]})/../logs/
-    log_dir=$(realpath $log_dir)
-    local timetag=$(date +%F"-"%T)"-"$RANDOM
-    local cur_pwd=$PWD
+    local case_name log_dir timetag
+    case_name=$(basename "$SCRIPT_NAME")
     case_name=${case_name%.*}
-    local case_folder=$log_dir/$case_name
-    mkdir -p $case_folder/$timetag
+    log_dir=$SCRIPT_HOME/logs/$case_name
```
Comment (marc-hb) [inline case-lib/logging_ctl.sh:55]: While quoting is not necessary here it would be good for consistency.

File: case-lib/opt.sh [hunk 6]
```diff
@@ -77,9 +79,10 @@ func_opt_parse_option()
                 if [ "${OPT_VALUE_lst[$i]}" ]; then
                     if [ "${OPT_PARM_lst[$i]}" -eq 1 ]; then
                         echo -e "\t""Default Value: ${OPT_VALUE_lst[$i]}"
+                    elif [ "${OPT_VALUE_lst[$i]}" -eq 0 ]; then
+                        echo -e "\t""Default Value: Off"
                     else
-                        [ "${OPT_VALUE_lst[$i]}" -eq 0 ] && echo -e "\t""Default Value: Off" \
-                            || echo -e "\t""Default Value: On"
+                        echo -e "\t""Default Value: On"
```
Comment (marc-hb) [inline case-lib/opt.sh:85]: Avoid `echo`, use `printf`
```shell
printf '\tDefault Value: '
if [ "${OPT_PARM_lst[$i]}" -eq 1 ]; then
   printf '%s\n' "${OPT_VALUE_lst[$i]}"
...
   printf 'On\n'
```

File: case-lib/pipeline.sh [hunk 7]
```diff
@@ -4,19 +4,15 @@ func_pipeline_export()
 {
     # no parameter input the function
     if [ $# -lt 1 ]; then
-        dlogi "Topology file name is not specified, unable to run command: ${BASH_SOURCE[-1]}" && exit 1
+        dlogi "Topology file name is not specified, unable to run command: $SCRIPT_NAME"
+        exit 1
```
Comment (marc-hb) [inline case-lib/pipeline.sh:8]: `die()`

File: case-lib/pipeline.sh [hunk 8]
```diff
@@ -4,19 +4,15 @@ func_pipeline_export()
...
-    tplg_path=$(func_lib_get_tplg_path "$tplg")
-    [[ "$?" != "0" ]] && dloge "No available topology for pipeline export" && exit 1
-    dlogi "${BASH_SOURCE[-1]} will use topology $tplg_path to run the test case"
-
-    # acquire filter option
-    local filter="$2"
+    tplg_path=$(func_lib_get_tplg_path "$1") || {
+        dloge "No available topology for pipeline export"
+        exit 1
+    }
```
Comment (marc-hb) [inline case-lib/pipeline.sh:14]: `die()`

File: case-lib/pipeline.sh [hunk 9]
```diff
@@ -28,18 +24,23 @@ func_pipeline_export()
...
-    local opt="-f \"$filter\" -b \"$ignore\""
-    local cmd="sof-tplgreader.py $tplg_path $opt -s $sofcard -e"
+    local opt=""
+    # acquire filter option
+    [[ "$2" ]] && opt="-f '$2' "
+    [[ "$ignore" ]] && opt="-b '$ignore' "
+    [[ "$SOFCARD" ]] && opt="-s $SOFCARD"
+
+    local -a pipeline_lst
+    local cmd="sof-tplgreader.py $tplg_path $opt -e" line=""
```
Comment (marc-hb) [inline case-lib/pipeline.sh:34]: This looks like real change, really not just fixing a warning. Explain this in a different commit or even better in a different PR.

File: case-lib/pipeline.sh [hunk 10]
```diff
@@ -48,5 +49,6 @@ func_pipeline_parse_value()
     local idx=$1
     local key=$2
     [[ $idx -ge $PIPELINE_COUNT ]] && echo "" && return
-    eval echo "\${PIPELINE_$idx['$key']}"
+    # shellcheck disable=SC1087
+    eval echo "\${PIPELINE_$idx['$key']}" # dynmaic echo the target value of the PIPELINE
```
Comment (marc-hb) [inline case-lib/pipeline.sh:53]: Using an intermediate variable (with a good name) should get rid of SC1087 while having simpler quoting.

**Comment by marc-hb** [comment]
About the [WIP] : if you submit the PR as a "github draft" then no reviewer will be automatically added.
https://github.blog/2019-02-14-introducing-draft-pull-requests/

**Comment by paulstelian97** [comment]
> It's too late now, you can't go back to Draft. Next time.

Not too late apparently. It can always be converted back and review requests re-sent.

**Comment by Bin-QA** [comment]
@marc-hb  
1. `die()` function to fix I want to add into other PR,  
it also refer the problem we have different `exit code`  
so here `die()` just focus on `dloge + exit 1`? 
2. `echo` replace with `print` just for the `opt.sh` or other file

Already verify on CI system

**Comment by marc-hb** [comment]
1. I keep losing track of where `die()` is available versus not. If it's not available here yet then forget it for now.

2. For very simple strings with no control characters `echo` is IMHO not the best but acceptable. For anything more advanced like for instance control characters please use `printf`.

**Comment by Bin-QA** [comment]
> 2. For very simple strings with no control characters `echo` is IMHO not the best but acceptable. For anything more advanced like for instance control characters please use `printf`.

@marc-hb I already try with `printf`
but I find, if I want use this style string `"description need two\nlines"` will not output
like my design, so it need to parse the string, how to deal those string for output

**Comment by paulstelian97** [comment]
> but I find, if I want use this style string `"description need two\nlines"` will not output
> like my design, so it need to parse the string, how to deal those string for output

Use single quotes on that, otherwise the shell will eat the `\`. Or do double escaping.

**Comment by marc-hb** [comment]
Exactly: always single quote the format string:
```shell
printf 'format: \t%s\nstring: \t%d' "$arg1" "$arg2"
```

**Comment by Bin-QA** [comment]
@marc-hb Can you help me to review this patch, or you think what I need to update

**Comment by marc-hb** [comment]
> Next time please submit several smaller PRs, one at a time thanks!

It's not only impossible to review changes that large, it's also impossible to test them in a reasonable time. See for instance regression #266

**Pull request merged**